apps/mobile-wallet/src/features/ecosystem/DAppWebViewScreen.tsx

```diff
@@ -0,0 +1,74 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import * as Clipboard from 'expo-clipboard'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import WebView from 'react-native-webview'
+import styled from 'styled-components/native'
+
+import Screen, { ScreenProps } from '~/components/layout/Screen'
+import { useWalletConnectContext } from '~/contexts/walletConnect/WalletConnectContext'
+import AddToFavoritesButton from '~/features/ecosystem/AddToFavoritesButton'
+import { activateAppLoading, deactivateAppLoading } from '~/features/loader/loaderActions'
+import { useAppDispatch } from '~/hooks/redux'
+import RootStackParamList from '~/navigation/rootStackRoutes'
+import { HEADER_OFFSET_TOP } from '~/style/globalStyle'
+
+interface DAppWebViewScreenProps extends NativeStackScreenProps<RootStackParamList, 'DAppWebViewScreen'>, ScreenProps {}
+
+const DAppWebViewScreen = ({ navigation, route, ...props }: DAppWebViewScreenProps) => {
+  const { dAppUrl, dAppName } = route.params
+
+  useDetectWCUrlInClipboardAndPair()
+
+  if (!dAppUrl) return null
+
+  return (
+    <Screen
+      titleAlwaysVisible
+      headerOptions={{
```

Comment:
The header is cut, but it's okay, it's a mvp, let's polish that later on.

![Uploading image.png…]()